app/src/main/java/br/com/catbag/gifreduxsample/models/AppState.java

```diff
@@ -23,4 +32,40 @@ public boolean getHasMoreGifs() {
         return true;
     }
 
-}
+    public String toJson() throws JsonProcessingException {
+        return new ObjectMapper().writeValueAsString(this);
+    }
+
+    public static AppState fromJson(String json) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new GuavaModule());
+        return mapper.readValue(json, AppState.class);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof AppState))
+            return false;
+        if (obj == this)
+            return true;
+
+        AppState state = (AppState) obj;
+        if (state.getGifs() == null)
+            return false;
+        if (this.getGifs().size() != state.getGifs().size())
+            return false;
+
+        for (String key : this.getGifs().keySet()) {
+            if (!state.getGifs().containsKey(key)) return false;
+            if (!state.getGifs().get(key).equals(this.getGifs().get(key))) return false;
+        }
+
+        return true;
+    }
```

Comment:
If is the same why override? Finally is or isnt the same? The quality folder contains pmd supression file, but we have the annotation option too.